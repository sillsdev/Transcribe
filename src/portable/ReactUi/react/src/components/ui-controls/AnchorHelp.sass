.AnchorHelp 
	padding: 0 20px
	width: 48px
	height: 48px
	text-align: center
