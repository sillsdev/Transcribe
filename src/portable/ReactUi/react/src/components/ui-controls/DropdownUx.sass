@import "../../_variables.sass"

.DropdownUx .css-10nd86i 
	width: 200px 

.DropdownUx .custom-select 
	position: relative
	font-family: $sans
	background: transparent
	color: $litUpText
	border: 1px solid $black
	border-radius: 4px
	font-size: 12pt
	select 
		display: none

  /*hide original SELECT element:*/
  .DropdownUx .select-selected 
    background-color: $black
  
  /*style the arrow inside the select element:*/
  .DropdownUx .select-selected:after 
    position: absolute
    content: ""
    top: 14px
    right: 10px
    width: 0
    height: 0
    border: 6px solid transparent
    border-color: #000 transparent transparent transparent
    border-radius: 4px
  
  /*point the arrow upwards when the select box is open (active):*/
  .DropdownUx .select-selected.select-arrow-active:after 
    border-color: transparent transparent #000 transparent
    border-radius: 4px
    top: 7px
  
  /*style the items (options), including the selected item:*/
  .DropdownUx .select-items div,.select-selected 
    color: $litUpText
    padding: 8px 16px
    border: 1px solid transparent
    border-color: transparent transparent rgba(0, 0, 0, 0.1) transparent
    border-radius: 4px
    cursor: pointer
    user-select: none
  
  /*style items (options):*/
  .DropdownUx .select-items 
    position: absolute
    background-color: DodgerBlue
    top: 100%
    left: 0
    right: 0
    z-index: 99
  
  /*hide the items when the select box is closed:*/

.DropdownUx .select-selected 
	background-color: $black
	&:after 
		position: absolute
		content: ""
		top: 14px
		right: 10px
		width: 0
		height: 0
		border: 6px solid transparent
		border-color: #000 transparent transparent transparent
		border-radius: 4px
	
	&.select-arrow-active:after 
		border-color: transparent transparent #000 transparent
		border-radius: 4px
		top: 7px
	

.DropdownUx .select-items div, .select-selected 
	color: $litUpText
	padding: 8px 16px
	border: 1px solid transparent
	border-color: transparent transparent rgba(0, 0, 0, 0.1) transparent
	border-radius: 4px
	cursor: pointer
	user-select: none

.DropdownUx .select-items 
	position: absolute
	background-color: DodgerBlue
	top: 100%
	left: 0
	right: 0
	z-index: 99

.DropdownUx .select-hide 
	display: none

.DropdownUx .select-items div:hover, .same-as-selected 
	background-color: rgba(0, 0, 0, 0.1)

.DropdownUx select option 
	margin: 10px 0px
	background: rgba(0, 0, 0, 0.3)
	color: $litUpText
	text-shadow: 0 1px 0 rgba(0, 0, 0, 0.4)

