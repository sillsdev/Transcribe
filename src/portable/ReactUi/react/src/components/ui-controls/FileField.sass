@import "../../_variables.sass"

.FileField
    width: 320px
    height: 56px !important
    display: flex
    flex-direction: column
    background: transparent
    justify-content: flex-start
    padding-bottom: 24px

    .bodyRow
        display: flex
        flex-direction: row
        background-color: rgba(255, 255, 255, 0.15)
        padding: 9px 12px 0 12px
        min-height: min-content
        justify-content: space-between

    .dataColumn
        display: flex
        flex-direction: column
        flex: 1 0 auto
        max-width: 264px
        overflow: hidden

    .clearButton
        display: flex
        flex: 0 0 auto

    .Caption
        font-family: $sans
        font-size: 12px
        height: 16px
        letter-spacing: 0.4px
        line-height: 16px

    input
        display: none

    .fileInput
        display: flex
        flex-direction: row
        flex: 1 0 auto
        height: 28px
        background: rgb(42, 82, 90)
        border: none
        visibility: hidden
        &::before
            visibility: visible
            display: flex
            flex-direction: row
            flex: 1 0 auto
            height: 28px
            content: attr(data-content)
            font-family: $sans
            font-size: 16px
            letter-spacing: 0.44px
    .DataColor
        color: $base2
    .CaptionGrey
        color: $base1

    .Message
        font-family: $sans
        font-size: 12px
        height: 16px
        letter-spacing: 0.4px
        line-height: 16px
        padding: 0 12px

    .BorderTopGreen
        border-top: 2px solid $greenLight

    .BorderTopRed
        border-top: 2px solid $red

    .CaptionGreen
        color: $greenLight

    .CaptionRed
        color: $red

    .Hide
        visibility: hidden
