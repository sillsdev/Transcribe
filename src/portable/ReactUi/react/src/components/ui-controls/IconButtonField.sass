@import "../../_variables.sass"

.IconButtonField
    height: 48px
    display: flex
    justify-content: center
    align-items: center
    cursor: pointer
    border: 1px solid transparent
    .Caption
        display: flex
        flex-direction: column
        justify-content: center
        width: 128px
        height: 48px
        font-family: $sans
        font-style: normal
        font-weight: 500
        line-height: normal
        font-size: 14px
        letter-spacing: 0.75px
        text-transform: uppercase
        color: $base2
    .CaptionImage
        margin: auto 16px
    .Reverse
        transform: scale(-1)

.DefaultBgColor
    background: $cyan3
    
.IconButtonField:hover
    background: $cyan3
    border: 1px solid $cyan2

.Hidden
    visibility: hidden

.NewSize
    width: 180px

.DisableText
    text-decoration: line-through
    color: #657B83
 


    