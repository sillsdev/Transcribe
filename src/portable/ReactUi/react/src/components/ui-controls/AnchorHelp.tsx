import * as React from 'react';
import './AnchorHelp.sass';

interface IProps {
    id: string;
    onClick: (value: string) => any;
};

class AnchorHelp extends React.Component<IProps, object> {
    constructor(props: IProps) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
    }

    public handleClick(event: any) {
        this.setState({ ...this.state, current: event.target.value })
        if (this.props.onClick != null) {
            this.props.onClick(event.target.value)
        }
    }

    public render() {
        const { id } = this.props
        return (
            <div id={id} className="AnchorHelp">
                <span>
                    <a key={"a"} className="linkText" onClick={this.handleClick}><img src={"/assets/QuestionGrey.svg"} alt="?" /></a>
                </span>
            </div>
        )
    }
};

export default AnchorHelp;
