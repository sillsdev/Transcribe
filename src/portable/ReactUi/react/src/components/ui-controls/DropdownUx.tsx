import * as React from 'react';
import './DropdownUx.sass';

interface IProps {
    collection: any[],
    id: string,
}
class DropdownUx extends React.Component<IProps, any> {
    public selected: string;
    private selectRef: any;

    constructor(props: IProps) {
        super(props)
        this.selectRef = React.createRef();
    }

    public render() {
        const options = this.props.collection.map((value: any) =>
            <option key={value} style={{backgroundColor: '#002B36', color: 'white'}} value={value}>{value}</option>
        )
        const doselect = () => this.onselect(this)
        this.selected = this.props.collection[0];
        return (
            <div id={this.props.id} className="DropdownUx">
                <select ref={this.selectRef} className="custom-select" onMouseUp={doselect}>
                   {options}
                </select>
            </div>
        )
    }

    private onselect(ctx: DropdownUx) {
        const r = ctx.selectRef.current
        ctx.selected = r && r.options[r.selectedIndex].text
    }
};

export default DropdownUx;

