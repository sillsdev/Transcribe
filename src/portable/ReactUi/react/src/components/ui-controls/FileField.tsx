import * as React from 'react';
import './FileField.sass';

interface IProps {
    id?: string;
    caption: string;
    inputValue?: string;
    extensions?: string;
    message?: string;
    isReadOnly?: boolean;
    onChange?: (name: string, data: string) => any;
}

class FileField extends React.Component<IProps, any> {
    public state = {
        current: this.props.inputValue,
        data: "",
    }

    constructor(props: IProps) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
        this.handleClear = this.handleClear.bind(this);
    }

    public handleChange(event: any) {
        let val: string = event.target.value;
        val = val.replace("C:\\fakepath\\", "")
        const reader = new FileReader()
        reader.readAsDataURL(event.target.files[0])
        reader.onload = (e: any) => {
            this.setState({...this.state, data: e.target.result})
            if (this.props.onChange != null) {
                this.props.onChange(val, e.target.result)
            }
        }
        this.setState({current: val });
    }

    public handleClear(event: any) {
        if (!this.props.isReadOnly) {
            this.setState({current: "" })
        }
        if (this.props.onChange != null) {
            this.props.onChange("", "")
        }
    }

    public render() {
        const { current } = this.state;
        const { extensions, caption, id, isReadOnly, message } = this.props;
        const errorMessage = current !== "" && message? message: "";
        const messageStyle = (errorMessage.length > 0)? "Message CaptionRed BorderTopRed": "Message BorderTopGreen";
        const captionStyle = (errorMessage.length > 0)? "Caption CaptionRed": "Caption CaptionGreen";
        const clearStyle = isReadOnly || current === ""? "Hide": "clearButton";
        const labelContent = (current && current !== "")? current : caption
        const labelStyle = (current && current !== "")? "DataColor" : "CaptionGrey"
        const accept = extensions != null? extensions: ".mp3,.wav";

        return (
            <div id={id} className="FileField">
                <div className="bodyRow">
                    <div className="dataColumn">
                        <label
                            className={captionStyle}>
                            {current && current !== ""? caption: ""
                        }</label>
                        <label
                            className={"fileInput " + labelStyle}
                            data-content={labelContent}
                            htmlFor="Upload"/>
                        <input
                            id="Upload"
                            readOnly={isReadOnly}
                            type="file"
                            accept={accept}
                            onChange={this.handleChange}/>
                    </div>
                    <div className={clearStyle} onClick={this.handleClear}>
                        <img src="/assets/Clear.svg" alt="X" />
                    </div>
                </div>
                <div className={messageStyle}>
                    <label>{errorMessage}</label>
                </div>
            </div>
        )
    }
}

export default FileField;