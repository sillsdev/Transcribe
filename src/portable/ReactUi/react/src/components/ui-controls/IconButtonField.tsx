import * as React from 'react';
import './IconButtonField.sass';

interface IProps {
    id?: string;
    caption: string;
    imageUrl: string;
    bgColor?: boolean;
    reverse?: boolean,
    hidden?: boolean,
    changeCaptionSize?: boolean,
    enabled?: boolean,
    onClick?: () => any;
}

export class IconButtonField extends React.Component<IProps, any> {

    constructor(props: IProps) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
    };

    public render() {

        const { bgColor, caption, changeCaptionSize, enabled, hidden, id, imageUrl, reverse } = this.props;
        let iconButtonStyle = (bgColor) ? "IconButtonField DefaultBgColor" : "IconButtonField"
        iconButtonStyle = (hidden)? iconButtonStyle + " Hidden": iconButtonStyle
        iconButtonStyle = (changeCaptionSize)?  iconButtonStyle + " NewSize": iconButtonStyle
        const imageStyle = (reverse)? "CaptionImage Reverse": "CaptionImage"
        const textStyle = (enabled === undefined)? "" : (enabled)? "": " DisableText";
        return (
            <div id={id} className={iconButtonStyle} onClick={this.handleClick}>
                <img src={"/assets/" + imageUrl} className={imageStyle} alt={caption && caption.charAt(0).toUpperCase()} />
                <label className={"Caption" + textStyle}>{caption && caption.toUpperCase()}</label>
            </div>
        )
    };

    private handleClick() {
        if (this.props.onClick != null) {
            this.props.onClick()
        }
    };
};

export default IconButtonField;