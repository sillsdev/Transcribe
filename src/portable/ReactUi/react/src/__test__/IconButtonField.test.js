/* jshint esversion: 6 */
import React from 'react';
import expect from 'expect';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import {IconButtonField} from '../components/ui-controls/IconButtonField';

configure({ adapter: new Adapter() });

// Snapshot for IconButtonField
describe('>>>Control: IconButtonField --- Snapshot',()=>{
    const minProps = {
        caption: "Caption",
        imageUrl: "cancel.svg",
    };

    it('+++capturing Snapshot of IconButtonField', () => {
        const renderedValue =  renderer.create(<IconButtonField {...minProps}/>).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });
});
//*******************************************************************************************************
describe('>>>Control: IconButtonField', () => {
    let wrapper;
    const minProps = {
        caption: "Caption",
        imageUrl: "cancel.svg"
    };

    beforeEach(() => {
        wrapper = shallow(<IconButtonField {...minProps} />);
    });

    it('+++ renders without exploding', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('+++ activates icon button link action', () => {
        const onButtonClick = sinon.spy();
        const linkWrapper = shallow(<IconButtonField {...minProps} id="myButton" onClick={onButtonClick}/>);
        linkWrapper.find('#myButton').simulate('click');
        expect(onButtonClick.calledOnce).toEqual(true);
    });
});
