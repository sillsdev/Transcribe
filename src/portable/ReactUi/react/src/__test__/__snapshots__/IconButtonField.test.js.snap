// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Control: IconButtonField --- Snapshot +++capturing Snapshot of IconButtonField 1`] = `
<div
  className="IconButtonField"
  id={undefined}
  onClick={[Function]}
>
  <img
    alt="C"
    className="CaptionImage"
    src="/assets/cancel.svg"
  />
  <label
    className="Caption"
  >
    CAPTION
  </label>
</div>
`;
